csharp/src/Ice/Exception.cs

```diff
@@ -8,40 +8,29 @@
 
 namespace IceInternal
 {
-    public class Ex
+    internal class Ex
     {
-        public static void ThrowUOE(Type expectedType, Ice.AnyClass v)
+        // UOE = UnknownObjectException as in unknow class. TODO: fix/remove.
+        internal static void ThrowUOE(Type expectedType, Ice.AnyClass v)
         {
-            //
             // If the object is an unknown sliced object, we didn't find an
             // class factory, in this case raise a NoClassFactoryException
             // instead.
-            //
             if (v is Ice.UnknownSlicedClass usv)
             {
-                throw new Ice.NoClassFactoryException("", usv.ice_id());
+                throw new Ice.NoClassFactoryException("", usv.TypeId ?? "");
             }
 
-            string type = v.ice_id();
-            string expected;
-            try
-            {
-                expected = (string)expectedType.GetMethod("ice_staticId").Invoke(null, null);
-            }
-            catch (Exception)
-            {
-                expected = "";
-                Debug.Assert(false);
-            }
-
-            throw new Ice.UnexpectedObjectException("expected element of type `" + expected + "' but received `" +
-                                                    type + "'", type, expected);
+            string typeId = Ice.TypeExtensions.GetIceTypeId(v.GetType())!;
```

Comment:
Why not call it as an extension method here?

Reply:
It's in the IceInternal namespace and this code is going to be removed soon.